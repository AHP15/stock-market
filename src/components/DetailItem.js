import PropTypes from 'prop-types';

import more from '../assets/arrow.png';

import styles from '../styles/DetailItem.module.css';

const DetailItem = ({ title, value, dark }) => (
  <div className={dark ? styles.dark : styles.item}>
    <p className={styles.title}>{title}</p>
    <p>{value}</p>
    <img src={more} alt="arrow" />
  </div>
);

DetailItem.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  dark: PropTypes.bool,
};

DetailItem.defaultProps = {
  dark: false,
};
export default DetailItem;
