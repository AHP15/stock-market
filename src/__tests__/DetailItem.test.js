import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import DetailItem from '../components/DetailItem';

it('renders DetailItem correctly', () => {
  const tree = renderer.create(<DetailItem title="test" value="10" />).toJSON();
  expect(tree).toMatchSnapshot();
});
