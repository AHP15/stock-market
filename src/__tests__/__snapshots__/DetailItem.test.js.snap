// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DetailItem correctly 1`] = `
<div
  className="item"
>
  <p
    className="title"
  >
    test
  </p>
  <p>
    10
  </p>
  <img
    alt="arrow"
    src="arrow.png"
  />
</div>
`;
